--- a/src/main/scala/nutcore/backend/seq/ISU.scala	2026-05-15 14:25:04.082587277 +0800
+++ b/src/main/scala/nutcore/backend/seq/ISU.scala	2026-05-15 14:26:13.999995746 +0800
@@ -41,7 +41,7 @@
   def isDepend(rfSrc: UInt, rfDest: UInt, wen: Bool): Bool = (rfSrc =/= 0.U) && (rfSrc === rfDest) && wen
 
   val forwardRfWen = io.forward.wb.rfWen && io.forward.valid
-  val dontForward1 = (io.forward.fuType =/= FuType.alu) && (io.forward.fuType =/= FuType.lsu)
+  val dontForward1 = true.B
   val src1DependEX = isDepend(rfSrc1, io.forward.wb.rfDest, forwardRfWen)
   val src2DependEX = isDepend(rfSrc2, io.forward.wb.rfDest, forwardRfWen)
   val src1DependWB = isDepend(rfSrc1, io.wb.rfDest, io.wb.rfWen)
