--- a/src/main/scala/nutcore/backend/fu/ALU.scala	2026-05-15 14:24:50.646316586 +0800
+++ b/src/main/scala/nutcore/backend/fu/ALU.scala	2026-05-15 14:26:14.011995989 +0800
@@ -123,7 +123,7 @@
   Debug(valid && (io.cfIn.instr(1,0) === "b11".U) =/= !isRVC, "[ERROR] pc %x inst %x rvc %x\n",io.cfIn.pc, io.cfIn.instr, isRVC)
   io.redirect.target := Mux(!taken && isBranch, Mux(isRVC, io.cfIn.pc + 2.U, io.cfIn.pc + 4.U), target)
   // with branch predictor, this is actually to fix the wrong prediction
-  io.redirect.valid := valid && isBru && predictWrong
+  io.redirect.valid := valid && isBru && predictWrong && !isBranch
   val redirectRtype = if (EnableOutOfOrderExec) 1.U else 0.U
   io.redirect.rtype := redirectRtype
   // mark redirect type as speculative exec fix
