--- a/src/main/scala/nutcore/backend/seq/EXU.scala	2026-05-11 13:36:13.405721146 +0800
+++ b/src/main/scala/nutcore/backend/seq/EXU.scala	2026-05-15 14:35:31.095214906 +0800
@@ -87,7 +87,7 @@
 
   io.out.bits.decode := DontCare
   (io.out.bits.decode.ctrl, io.in.bits.ctrl) match { case (o, i) =>
-    o.rfWen := i.rfWen && (!lsuTlbPF && !lsu.io.loadAddrMisaligned && !lsu.io.storeAddrMisaligned || !fuValids(FuType.lsu)) && !(csr.io.wenFix && fuValids(FuType.csr))
+    o.rfWen := i.rfWen && !(csr.io.wenFix && fuValids(FuType.csr))
     o.rfDest := i.rfDest
     o.fuType := i.fuType
   }
