diff --git a/src/main/scala/nutcore/backend/fu/MDU.scala b/src/main/scala/nutcore/backend/fu/MDU.scala
index f50b44a..8b69604 100644
--- a/src/main/scala/nutcore/backend/fu/MDU.scala
+++ b/src/main/scala/nutcore/backend/fu/MDU.scala
@@ -87,7 +87,7 @@ class Divider(len: Int = 64) extends NutCoreModule {
   val (aSign, aVal) = abs(a, io.sign)
   val (bSign, bVal) = abs(b, io.sign)
   val aSignReg = RegEnable(aSign, newReq)
-  val qSignReg = RegEnable((aSign ^ bSign) && !divBy0, newReq)
+  val qSignReg = RegEnable((aSign ^ bSign), newReq)
   val bReg = RegEnable(bVal, newReq)
   val aValx2Reg = RegEnable(Cat(aVal, "b0".U), newReq)
 
@@ -106,7 +106,7 @@ class Divider(len: Int = 64) extends NutCoreModule {
     // When divide by 0, the quotient should be all 1's.
     // Therefore we can not shift in 0s here.
     // We do not skip any shift to avoid this.
-    cnt.value := Mux(divBy0, 0.U, Mux(canSkipShift >= (len-1).U, (len-1).U, canSkipShift))
+    cnt.value := Mux(canSkipShift >= (len-1).U, (len-1).U, canSkipShift)
     state := s_shift
   } .elsewhen (state === s_shift) {
     shiftReg := aValx2Reg << cnt.value
