--- a/src/main/scala/nutcore/backend/fu/CSR.scala	2026-05-11 13:36:13.405721146 +0800
+++ b/src/main/scala/nutcore/backend/fu/CSR.scala	2026-05-15 14:26:14.019996150 +0800
@@ -684,7 +684,7 @@
     val mstatusNew = WireInit(mstatus.asTypeOf(new MstatusStruct))
     // mstatusNew.mpp.m := ModeU //TODO: add mode U
     mstatusNew.ie.m := mstatusOld.pie.m
-    privilegeMode := mstatusOld.mpp
+    privilegeMode := ModeU
     mstatusNew.pie.m := true.B
     mstatusNew.mpp := ModeU
     mstatus := mstatusNew.asUInt
