--- a/src/main/scala/nutcore/mem/EmbeddedTLB.scala	2026-05-11 13:36:13.409721498 +0800
+++ b/src/main/scala/nutcore/mem/EmbeddedTLB.scala	2026-05-15 14:26:14.027996311 +0800
@@ -101,7 +101,7 @@
   // meta reset
   val flushTLB = WireInit(false.B)
   BoringUtils.addSink(flushTLB, "MOUFlushTLB")
-  mdTLB.reset := reset.asBool || flushTLB
+  mdTLB.reset := reset.asBool
 
   // VM enable && io
   val vmEnable = satp.asTypeOf(satpBundle).mode === 8.U && (io.csrMMU.privilegeMode < ModeM)
