diff --git a/src/main/scala/nutcore/isa/RVZifencei.scala b/src/main/scala/nutcore/isa/RVZifencei.scala
index 1eca5ff..6627351 100644
--- a/src/main/scala/nutcore/isa/RVZifencei.scala
+++ b/src/main/scala/nutcore/isa/RVZifencei.scala
@@ -20,7 +20,7 @@ import chisel3._
 import chisel3.util._

 object RVZifenceiInstr extends HasInstrType {
-  def FENCEI = BitPat("b000000000000_00000_001_00000_0001111")
+  def FENCEI = BitPat("b????????????_?????_001_?????_0001111")

   val table = Array(
     FENCEI -> List(InstrB, FuType.mou, MOUOpType.fencei)
