--- a/src/main/scala/nutcore/backend/fu/CSR.scala	2026-05-17 10:00:00.000000000 +0800
+++ b/src/main/scala/nutcore/backend/fu/CSR.scala	2026-05-17 10:01:00.000000000 +0800
@@ -869,7 +869,7 @@
-  when(perfCntCond(0xb03 & 0x7f)) { perfCnts(0xb02 & 0x7f) := perfCnts(0xb02 & 0x7f) + 2.U } // Minstret += 2 when MultiCommit
+  when(true.B) { perfCnts(0xb02 & 0x7f) := perfCnts(0xb02 & 0x7f) + 1.U } // Minstret += 2 when MultiCommit
